### ISSUE TITLE
TypeError: undefined is not an object (evaluating 'this._conversationsPromise.then')

### ISSUE THREAD

**Issue by Reeywhaar** [twilio/twilio-conversations-demo-react#120]
We get a lot of errors that `this._conversationsPromise` is undefined when trying to call `getSubscribedConversations` on a channel. Let's take a look at `getSubscribedConversations`:
```
async getSubscribedConversations() {
    await this._ensureReady;
    return this._conversationsPromise.then((conversations) => conversations.getConversations());
}
```

Let's look how `_ensureReady` is defined in `client.js` of `@twilio/conversations`
```
this._ensureReady = new Promise((resolve, reject) => {
    this._resolveEnsureReady = resolve;
    this._rejectEnsureReady = reject;
 }).catch(() => void 0); // @todo How to process unhandled rejection here?
```

Well, it seems that there is a bug here and `await this._ensureReady` will never throw because you swallowed the error. Since you swallowed the error we get to the next line `return this._conversationsPromise...`. But we can't be sure that `_conversationsPromise` was set.

You should remove that `catch` on `_ensureReady` in Client constructor. At least users or your library would get the original error.

**Comment by berkus** [comment]
When you remove the catch, what is the error you get?

**Comment by Reeywhaar** [comment]
Also I think you should reset `_ensureReady` when disconnecting. I mean set `_ensureReady` to `new Promise(...` again

**Comment by Reeywhaar** [comment]
> When you remove the catch, what is the error you get?

Didn't try yet but I suspect it is some connectivity error

**Comment by berkus** [comment]
Yep, ok, I will take a look at that part, thank you for reporting this!

**Comment by mmubasher** [comment]
Waiting for this fix

**Comment by subhabagchi** [comment]
Any update on this bug ?

**Comment by agrukhal** [comment]
> Any update on this bug ?

@subhabagchi Hi! Since the issue was opened a few updates were made. Is the problem still persists for you on newer versions of SDK?

**Comment by subhabagchi** [comment]
@agrukhal We will try and let you know.

**Comment by agrukhal** [comment]
Hi! I will close this issue for now. Please, open a new one with reproduction steps in case the issue persists. Thank you!
